crashappsec/chalk
# feat(cmd_login): add new chalk command for API `login` separate from key generation

## Issue

Proactive work stemming from #27

## Description

This adds a new chalk command `chalk login` that initiates the API login process, specifically an OIDC device code flow, separately from the overall `chalk setup` process. `chalk setup` will still prompt the user to go through the login process if it hasn't already been performed, but will not re-prompt a user to authenticate if they already have API tokens saved.

Prior to this command the receipt/refresh of API tokens was implicitly linked to the key generation and management that chalk setup performs. It makes sense to separate this functionality even if for now the API is only serving the purposes of key management.

## Testing

To show the separate processes at play you can run the following series of commands:

* Login to API : `chalk --trace login`
     * follow the URL and complete authentication
* Run key setup: `chalk --trace setup gen`

Note that the `setup` command does not prompt the user for authentication. If it is run without the proceeding call to `login` then the user will be prompted to authenticate.
